frontend/apps/reader/modules/readerpaging.lua

```diff
@@ -1029,8 +1029,15 @@ function ReaderPaging:onGotoPageRel(diff)
             goto_end(y)
             goto_end(x)
         elseif new_page > 0 then
-            self:_gotoPage(new_page)
-            goto_end(y, -y_diff)
+            -- next page
+            if y_diff == 1 then
+                goto_end(y, -y_diff)
+                self:_gotoPage(new_page)
+            -- previous page
+            else
+                self:_gotoPage(new_page)
+                goto_end(y, -y_diff)
+            end
```

Comment:
(Not really familiar with all that) but it feels like y_diff may not always be 1 or -1, it could be 0.3, no?
Or you are just fixing the case where y_diff == exactly 1 ? (as the other branch does what it did always before).
And does goto_end() before _gotoPage() has any effect ?
Still unsure what this does, I did not really understand your original issue :)

Aesthetics wise, your `-- previous page` comment screws a bit the reading of the indented blocks; maybe put them on the same lines with the `then -- next page` and `else -- previous page`.

Reply:
> (Not really familiar with all that) but it feels like y_diff may not always be 1 or -1, it could be 0.3, no? Or you are just fixing the case where y_diff == exactly 1 ? (as the other branch does what it did always before). And does goto_end() before _gotoPage() has any effect ? Still unsure what this does, I did not really understand your original issue :)

 I'm not familiar either, but became 1% more now to see if y_diff can be 0.3. onGotoPageRel is called with -1, 1 on taps.

```
    self.ui:registerTouchZones({
                if G_reader_settings:nilOrFalse("page_turns_disable_tap") then
                    return self:onGotoViewRel(1)
                end

                if G_reader_settings:nilOrFalse("page_turns_disable_tap") then
                    return self:onGotoViewRel(-1)

        
    function ReaderPaging:onGotoViewRel(diff)
        ...
        self:onGotoPageRel(diff)
```
        
        
        
At first glance looks like onGotoPageRel does with y_diff is only shifts signal:

```
    function ReaderPaging:onGotoPageRel(diff)
    ..
    local y_diff = diff

    if right_to_left then
        x_diff, y_diff = -x_diff, -y_diff
    end
    ...
```

My issue I create a test case on a book I was reading. You see in the issue description that when we go to next page the view advances more than the necessary and I need additional tap left to go back to the top of the image. It doesn't occur with all files, so right now is a fix for this specific case.

goto_end() before go to the next page seems to clear the page difference. 
In case of reverse, seems not to break bottom_to_top mode after some tests, but can add a check for if bottom_to_top then... if someone complain some kind of problem...

> 
> Aesthetics wise, your `-- previous page` comment screws a bit the reading of the indented blocks; maybe put them on the same lines with the `then -- next page` and `else -- previous page`.

Reply:
OK, I see it is only called with 1 or -1. (The question is if, even if we use it with only 1 or -1, it's supposed to work with a fractional value and we would be breaking it :) but it feels it has been made to only work with 1 or -1, so let's not bother.)

> when we go to next page the view advances more than the necessary and I need additional tap left to go back to the top of the image. It doesn't occur with all files, so right now is a fix for this specific case.

Would be good to figure why there is this unwanted shift, and why not always... (footer not accounted in the y-pan ? but it would then be always... rounding error somewhere, but 1px would hardly be noticable...)

> goto_end() before go to the next page _seems_ to clear the page difference.

Any explanation for that _seems_ ? :)

```lua
            if y_diff == 1 then -- next page
                goto_end(y, -y_diff)
                self:_gotoPage(new_page) 
```

I would naively think this goto_end is a no-op. But may be it updates the view area stuff according to the geometry of the current page, and then going to the next page, using the geometry/viewarea of the previous page, which feels odd :)

Could the reason for "doesn't occur with all files" be that the files where it happens don't have all their pages the same size ? And you get the shifts only when crossing pages of different heights ?
It still feels that going to the next page, and then adjusting view area, is the good logical ordering of actions :/

Does this also work when the zoom direction is bottom to top or right to left ? When going to a next page, are we rightly _not in_ the top left quadrant ?

Just asking - so you can get 3% more familiar :)
(and may be in 25 more such back-and-forths between us, you'll be 100% familiar with all that ! :))

Reply:
> OK, I see it is only called with 1 or -1. (The question is if, even if we use it with only 1 or -1, it's supposed to work with a fractional value and we would be breaking it :) but it feels it has been made to only work with 1 or -1, so let's not bother.)
> 
> > when we go to next page the view advances more than the necessary and I need additional tap left to go back to the top of the image. It doesn't occur with all files, so right now is a fix for this specific case.
> 
> Would be good to figure why there is this unwanted shift, and why not always... (footer not accounted in the y-pan ? but it would then be always... rounding error somewhere, but 1px would hardly be noticable...)

Believe or not I created a test case (all pages same size) when footer disabled all things looks good and when enabled the footer I face the same problem with a minor degree, the footer size. Maybe messed with crop or something. But thankfully I deleted this test case and I can't reproduce or create similar test case anymore.

> 
> > goto_end() before go to the next page _seems_ to clear the page difference.
> 
> Any explanation for that _seems_ ? :)
> 
> ```lua
>             if y_diff == 1 then -- next page
>                 goto_end(y, -y_diff)
>                 self:_gotoPage(new_page) 
> ```
> 
> I would naively think this goto_end is a no-op. But may be it updates the view area stuff according to the geometry of the current page, and then going to the next page, using the geometry/viewarea of the previous page, which feels odd :)
> 
> Could the reason for "doesn't occur with all files" be that the files where it happens don't have all their pages the same size ? And you get the shifts only when crossing pages of different heights ? It still feels that going to the next page, and then adjusting view area, is the good logical ordering of actions :/

I think it is, more or less in the right path now.
I made a test case all the three pages the same image and the problem doesn't occur. 
But based on this one I also made another test case with the first page height bigger, and another test case with the second image height bigger. The problem doesn't occur either. That leads that at least a additional tap need to be done before going to the next page and possibly my fix is forcing something like that.
Now I need to figure out the reason in terms of code.  
> 
> Does this also work when the zoom direction is bottom to top or right to left ? When going to a next page, are we rightly _not in_ the top left quadrant ?
> 
> Just asking - so you can get 3% more familiar :) (and may be in 25 more such back-and-forths between us, you'll be 100% familiar with all that ! :))